python/src/spark_rapids_ml/core.py

```diff
@@ -1041,7 +1061,69 @@ def _transform_udf(pdf_iter: Iterator[pd.DataFrame]) -> pd.DataFrame:
                         data[pred.model_index] = index
                     yield data
 
-        return dataset.mapInPandas(_transform_udf, schema=schema)  # type: ignore
+        df = dataset.mapInPandas(_transform_udf, schema=schema)  # type: ignore
+        return self._try_stage_level_scheduling(df)
+
+    def _try_stage_level_scheduling(self, df: DataFrame) -> DataFrame:
+        """Try to enable stage-level scheduling for transform tasks"""
+        if self._skip_stage_level_scheduling():
+            return df
+
+        ss = _get_spark_session()
+
+        # Default the concurrent transform tasks to 2
+        concurrent_transform_tasks = int(
+            ss.conf.get("spark.rapids.ml.concurrentPredictTasks", "2")  # type: ignore
+        )
+
+        if concurrent_transform_tasks <= 0:
+            raise ValueError(
+                "The value of spark.rapids.ml.concurrentPredictTasks must be greater than 0."
+            )
+
+        executor_cores = int(ss.sparkContext.getConf().get("spark.executor.cores"))  # type: ignore
+
+        if (
+            executor_cores // 2 < concurrent_transform_tasks < executor_cores
+            or concurrent_transform_tasks > executor_cores
+        ):
+            original_concurrent_transform_tasks = concurrent_transform_tasks
+            concurrent_transform_tasks = (
+                executor_cores // 2
```

Comment:
Is it possible to achieve exactly concurrent_transform_tasks by setting executor_cores=concurrent_transform_tasks and task_cores=1 just for this stage?   Or is executor_cores not configurable per stage?

Reply:
Actually, we can't require executor resources which will cause allocating new executors. But in our case, we don't want to require more executores.